### PR TITLE
Performance test uses driver

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

This PR adds a script to do performance test from driver. 

## Code changes:

- Added `driver_performance.sh` to collect performance json file

## Infrastructure changes:

- pace performance test should now use driver instead of dynamics.py

## Checklist
Before submitting this PR, please make sure:

- [ ] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] For each public change and fix in `pace-util`, HISTORY has been updated
- [ ] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: driver/pace/driver/driver.py [hunk 1]
```diff
@@ -598,11 +599,16 @@ def _critical_path_step_all(
     def step_all(self):
         logger.info("integrating driver forward in time")
         with self.performance_collector.total_timer.clock("total"):
+            self.profiler.enable()
```
Comment (elynnwu) [inline driver/pace/driver/driver.py:602]: It doesn't actually make sense to have this for dace orchestration, do we want to support this for other backends?

File: .jenkins/driver_performance.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#SBATCH --partition=normal
+########################################################
+set -x
+export OMP_NUM_THREADS=12
+export FV3_DACEMODE=BuildAndRun
+srun python -m pace.driver.run ${JENKINS_DIR}/driver_configs/baroclinic_c192_6ranks.yaml
+EOF
+launch_job run.daint.slurm 3600
+
+cat << EOF > results.py
```
Comment (twicki) [inline .jenkins/driver_performance.sh:39]: this should probably live somewhere and not be created on the fly, right?

File: driver/pace/driver/driver.py (hunk 1 above)
Comment (twicki) [reply driver/pace/driver/driver.py:602]: I would keep it here

File: util/pace/util/_profiler.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        self._enabled = False
+        self.profiler.dump_stats(filename)
+
+    @property
+    def enabled(self) -> bool:
+        """Indicates whether the profiler is currently enabled."""
+        return self._enabled
+
+
+class NullProfiler(Profiler):
```
Comment (twicki) [inline util/pace/util/_profiler.py:24]: why are we inheriting from the base-profiler here? This seems dangerous. If we want a base profiler it should be an abc, right?

**Review by twicki — CHANGES_REQUESTED** [review]

File: .jenkins/driver_performance.sh (hunk 2 above)
Comment (elynnwu) [reply .jenkins/driver_performance.sh:39]: Moved to be a separate file.

File: util/pace/util/_profiler.py (hunk 3 above)
Comment (elynnwu) [reply util/pace/util/_profiler.py:24]: Removed inheritance

**Review by twicki — APPROVED** [review]
lgtm

**Pull request merged**